invenio_resources/service/state.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Invenio-Records-Agent is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""State API."""
+
+
+class RecordState:
```

Comment:
Medium: `State` is probably too generic for us (every object has state, "state" could be used for anything...). My suggestion for the name: `IdentifiedRecord` . It represents a Record and its identifiers. It has an id and the PersistentIdentifiers mapping to it. It is identified.

This is a really good object as it solves our [data clump](https://en.wikipedia.org/wiki/Data_clump) problem. It should almost be passed around in most of Invenio IMHO.